lib/core/test/unit/Cardano/Wallet/Primitive/BlockSummarySpec.hs

```diff
@@ -28,32 +32,58 @@ import Test.QuickCheck
     ( Arbitrary (..)
     , Gen
     , Property
+    , forAll
     , listOf1
     , property
     , resize
     , shuffle
+    , sublistOf
     , (===)
     )
 
 import Data.Map.Strict as Map
 
 spec :: Spec
 spec = do
+    describe "Sublist" $ do
```

Comment:
I would also distinct the following properties: 
1. for any two sublists of the same list the result of merge is sublist that contains both sublists, is union sublist. 
2. merging is commutative

Reply:
Ok! There's not much that can go wrong in this code, but these laws are aesthetically pleasing. 😊

Reply:
I was about to write that property 2 follows from property 1 because set union is commutative, but property 2 is slightly stronger due to the order of elements. 🤔